### ISSUE TITLE
Стопается на 20и использованиях колышка

### ISSUE THREAD

**Issue by alfedukovich** [AlexGyver/alexgyver.github.io#26]
![изображение](https://user-images.githubusercontent.com/8695578/227466928-4b04a8c7-c8c2-45c5-9ddd-9ac22afdb130.png)
В этом случае похоже на то, что не дорисовано.
Имхо, лучше оставить параметр настраиваемым, как было раньше.

**Comment by alfedukovich** [comment]
Как насчёт того чтобы не останавливать рендеринг, а запрещать выбирать колышек на котором уже есть n-ое количество петель?

**Comment by GyverLibs** [comment]
Этот стоп происходит в режиме negative когда сумма лучшей нитки равна нулю. Если остановки не будет - рисование все равно остановится, но будет бегать между двумя соседними гвоздями

**Comment by GyverLibs** [comment]
> запрещать выбирать колышек на котором уже есть n-ое количество петель?

Это делает overlaps

**Comment by alfedukovich** [comment]
> Это делает overlaps

У меня рендеринг стопается на этом шаге.

**Comment by alfedukovich** [comment]
> Этот стоп происходит в режиме negative когда сумма лучшей нитки равна нулю. Если остановки не будет - рисование все равно остановится, но будет бегать между двумя соседними гвоздями

До обновления, дорисовывало в этом режиме и не циклилось)
![изображение](https://user-images.githubusercontent.com/8695578/227470638-7491270a-ac42-4b99-93ec-29ee41268f9a.png)

**Comment by GyverLibs** [comment]
Остановка происходит при включенном negative, когда сумма лучшей линии равна 0

**Comment by alfedukovich** [comment]
Понял что это не привязано к ovrlaps. 
Можешь разрешить суммам быть отрицательными для режима Negativ?)

**Comment by GyverLibs** [comment]
В том и дело что оно не будет отрицательным, оно падает до нуля и после этого рисование само по себе уже не продолжается, даже без условия остановки. Закомментируй это условие, там max == 0 или что то такое

**Comment by GyverLibs** [comment]
Что то менять будем?

**Comment by alfedukovich** [comment]
Проверку на max=0 нужно отключить при опции Negativ

**Comment by alfedukovich** [comment]
По хорошему бы еще понять - почему с Center Balance циклится?

**Comment by alfedukovich** [comment]
По сравнению с гитхабом есть изменения, кроме отключённого max==0?

**Comment by GyverLibs** [comment]
Я работаю сразу с мастер веткой)

**Comment by alfedukovich** [comment]
Можешь залить версию с отключённым max==0 на гит pages? Чтобы быть уверенными что один и тот-же код.

**Comment by alfedukovich** [comment]
И тест такой - стандартная двочка, количество линий 2000, Negativ и Quarter

**Comment by GyverLibs** [comment]
У меня нет такой версии)

**Comment by alfedukovich** [comment]
Вот сравнение - последний комит из гитхаба, и одно изменение - закоменченный max==0. Браузер Chrome
https://user-images.githubusercontent.com/8695578/227488454-690824d7-ced1-4176-8386-d1e35bf64965.mp4

**Comment by AlexGyver** [comment]
так, вот что происходит при Negative. Как только начинается подряд куча нулей - отрисовка фактически останавливается. В консоль выведено `console.log(max+','+best);` и видно что оно скачет между одними и теми же гвоздями
![image](https://user-images.githubusercontent.com/21090953/227499426-bc1ee84a-d07c-41f0-a378-4ba81e3edb07.png)

**Comment by AlexGyver** [comment]
сделал буфер на последние 5 максов, если они все нули - останавливаем, теперь рисует до упора
